#!/usr/bin/env python
PACKAGE = "apm"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("K_yaw_P", double_t, 0, "K_yaw_P", 5, 0, 150)
gen.add("K_yaw_I", double_t, 0, "K_yaw_I", 0, 0, 5)
gen.add("K_yaw_D", double_t, 0, "K_yaw_D", 0, 0, 15)

gen.add("K_x_P", double_t, 0, "K_x_P", 100, 0, 150)
gen.add("K_x_I", double_t, 0, "K_x_I", 0, 0, 50)
gen.add("K_x_D", double_t, 0, "K_x_D", 0, 0, 500)

gen.add("K_y_P", double_t, 0, "K_y_P", 100, 0, 150)
gen.add("K_y_I", double_t, 0, "K_I_P", 0, 0, 50)
gen.add("K_y_D", double_t, 0, "K_y_D", 0, 0, 500)

gen.add("K_z_P", double_t, 0, "K_z_P", 400, 0, 5000)
gen.add("K_z_I", double_t, 0, "K_z_I", 0, 0, 50)
gen.add("K_z_D", double_t, 0, "K_z_D", 0, 0, 500)

exit(gen.generate(PACKAGE, "apm_control", "control"))
